pandas/core/arrays/string_.py

```diff
@@ -144,11 +144,18 @@ class StringArray(PandasArray):
         .. warning::
 
            Currently, this expects an object-dtype ndarray
-           where the elements are Python strings or :attr:`pandas.NA`.
+           where the elements are Python strings
+           or nan-likes(``None``, ``nan``, ``NaT``, ``NA``, Decimal("NaN")).
```

Comment:
should we accept NaT or Decimal('NaN') as these are 'wrong' generally for this dtype.

Reply:
Will remove. (FWIW, they were just there because checknull recognizes them as null)